src/freenet/node/useralerts/N2NTMUserAlert.java

```diff
@@ -117,6 +117,11 @@ public FCPMessage getFCPMessage() {
 				composedTime, sentTime, receivedTime, messageText);
 	}
 
+    @Override
+    public long getUpdatedTime() {
+        return sentTime;
```

Comment:
It might be slightly more sensible to order by receive time: sent time is set by the sender, so does not necessarily lead to "newest first" behaviour (consider a friend being disconnected for some time).